lib/puppet/provider/cs_property/crm.rb

```diff
@@ -81,7 +81,37 @@ def flush
       # clear this on properties, in case it's set from a previous
       # run of a different corosync type
       ENV['CIB_shadow'] = nil
-      crm('configure', 'property', '$id="cib-bootstrap-options"', "#{@property_hash[:name]}=#{@property_hash[:value]}")
+      begin
+        crm('configure', 'property', '$id="cib-bootstrap-options"', "#{@property_hash[:name]}=#{@property_hash[:value]}")
+      rescue
+        xml = <<-EOF
+        <diff>
+          <diff-removed>
+            <cib>
+              <configuration>
+                <crm_config>
+                  <cluster_property_set id="cib-bootstrap-options">
+                    <nvpair value=#{@property_hash[:value].inspect} id=#{@property_hash[:name].inspect}/>
+                  </cluster_property_set>
+                </crm_config>
+              </configuration>
+            </cib>
+          </diff-removed>
+          <diff-added>
+            <cib>
+              <configuration>
+                <crm_config>
+                  <cluster_property_set id="cib-bootstrap-options">
+                    <nvpair value="stop" id="cib-bootstrap-options-no-quorum-policy"/>
+                  </cluster_property_set>
+                </crm_config>
+              </configuration>
+            </cib>
+          </diff-added>
+        </diff>
```

Comment:
@igalic 
I have already considered two options to generate XML "the normal way" - REXML and Builder::XmlMarkup. The first one makes the code even worse (you don't see the structure of the resulting xml document cleanly, just skimming the code), the second one makes the code much cleaner, but requires a gem "builder" to be installed as a dependency (which seems to be too much for just one place to generate xml).
IMHO inlined XML is good enough (there are a lot of manifests in the wild that inline shell scripts as well). However, if you and ffrank@ are totally against it, then I would prefer to go "Builder::XmlMarkup" way.